mlabs/test/Test/NFT/Contract.hs

```diff
@@ -170,3 +161,27 @@ testQueryOwner = check "Query owner" assertState w1 script
         nftId = NftId . hashData . mp'content $ artwork2
         owner = QueryCurrentOwner . Just . UserId . pubKeyHash . walletPubKey $ w1
         msg = queryCurrentOwnerLog nftId owner
+
+-- | User lists all NFTs in app
+testQueryListNfts :: TestTree
+testQueryListNfts = check "Query list NFTs" assertState w1 script
+  where
+    script = do
+      mapM_ (userMint w1) artworks
+      userQueryListNfts w1
+
+    assertState = assertInstanceLog (walletInstanceTag w1) (any predicate)
+
+    artworks = [artwork1, artwork2]
+
+    nfts =
+      sortOn info'id
+        . fmap (\mp -> mintParamsToInfo mp (UserId . pubKeyHash . walletPubKey $ w1))
+        $ artworks
+
+    predicate = \case
```

Comment:
As we are using this `predicate` to test all the queries, maybe it would be worth generalising it and adding it to ` mlabs/test/Test/NFT/Init.hs `.